service/operator/operatorkit.go

```diff
@@ -0,0 +1,146 @@
+package operator
+
+import (
+	microerror "github.com/giantswarm/microkit/error"
+)
+
+// Operator implements the building blocks of any operator business logic being
+// reconciled when observing TPRs. This interface provides a guideline for a
+// more easy way to follow the rather complex intentions of operators in
+// general.
+type Operator interface {
+	// GetCurrentState receives the custom object observed during TPR watches. Its
+	// purpose is to return the current state of the resources being managed by
+	// the operator. This can e.g. be some actual data within a configmap as
+	// provided by the Kubernetes API. This is not limited to Kubernetes resources
+	// though. Another example would be to fetch and return information about
+	// Flannel bridges.
+	GetCurrentState(obj interface{}) (interface{}, error)
+	// GetDesiredState receives the custom object observed during TPR watches. Its
+	// purpose is to return the desired state of the resources being managed by
+	// the operator. The desired state should always be able to be made up using
+	// the information provided by the TPO. This can e.g. be some data within a
+	// configmap, how it should be provided by the Kubernetes API. This is not
```

Comment:
Yeah it was actually hard to describe this. I see the confusion. This is about generating the desired state. In case an operator creates the desired state for some kubernetes resources `GetDesiredState` should return these resources exactly how they should be living inside Kubernetes. So when you lookup these resources the desired state describes how the Kubernetes API should provide these resources when you look them up. I am open to suggestions on how to improve the godoc here.